jcli/src/jcli_app/certificate/new_vote_cast.rs

```diff
@@ -1,44 +1,106 @@
 use crate::jcli_app::certificate::{write_cert, Error};
+use crate::jcli_app::utils;
+use bech32::FromBase32;
 use chain_impl_mockchain::{
     certificate::{Certificate, VoteCast, VotePlanId},
     vote::{Choice, Payload},
 };
+use rand_chacha::rand_core::SeedableRng;
 use std::path::PathBuf;
 use structopt::StructOpt;
 
-/// create a vote cast certificate
 #[derive(StructOpt)]
-pub struct VoteCastCmd {
+pub struct PublicVoteCast {
     /// the vote plan identified on the blockchain
-    pub vote_plan_id: VotePlanId,
+    vote_plan_id: VotePlanId,
 
     /// the number of proposal in the vote plan you vote for
-    pub proposal_index: u8,
+    proposal_index: u8,
 
     /// the number of choice within the proposal you vote for
-    pub choice: u8,
+    choice: u8,
+
+    /// write the output to the given file or print it to the standard output if not defined
+    #[structopt(long = "output")]
+    output: Option<PathBuf>,
+}
 
-    /// should be used for vote plans with public votes
-    #[structopt(long = "public")]
-    pub public: bool,
+#[derive(StructOpt)]
+pub struct PrivateVoteCast {
+    /// the vote plan identified on the blockchain
+    vote_plan_id: VotePlanId,
+
+    /// the number of proposal in the vote plan you vote for
+    proposal_index: u8,
+
+    /// size of voting options
+    options: usize,
+
+    /// the number of choice within the proposal you vote for
+    choice: u8,
+
+    /// key to encrypt the vote with
+    encrypting_key_path: Option<PathBuf>,
```

Comment:
Here too, I think, all fields should be options except the `<proposal_index> <choice>` 2D mnemonic.

We can leave these for a later bikeshedding PR if it's too much to do here; I've got some ideas :smiling_imp: